packages/utils/src/socialRemark/README.md

```diff
@@ -0,0 +1,84 @@
+# SocialRemark
+
+SocialRemark protocol is designed to process cross-chain actions.
+
+## SocialRemark message format
+
+- Domain Registration
+  `prot_name::version::action::op_id::target::domain_name::token`
+- Energy Generation
+  `prot_name::version::action::op_id::target::energy_amount::token`
+
+Remark message has 2 parts:
+- core part (is the same for all messages): `prot_name::version::action`
+- content part (depends on actions):
+  - `op_id::target::domain_name::token`
+  - `op_id::target::energy_amount::token`
+
+Core part options:
+- `prot_name` - protocol name (default valid value `social` but can be configured but `.setConfig()` method)
+- `version` - protocol version (current available version `0.1`)
+- `action` - message action (available actions: `DMN_REG`, `DMN_REG_OK`, `DMN_REG_REFUND`, `NRG_GEN`, `NRG_GEN_OK`, `NRG_GEN_REFUND`)
+
+Content options:
+- `op_id` - unique operation identified. Should be the same for all actions of one operation (the same `op_id` value for `DMN_REG` , `DMN_REG_OK` , `DMN_REG_REFUND` of A single domain registration flow. Necessary for relating actions with each other.)
```

Comment:
```suggestion
- `op_id` - A unique operation identifier. It should be the same for all actions of one operation (the same `op_id` value for `DMN_REG` , `DMN_REG_OK` , `DMN_REG_REFUND` of A single domain registration flow. Necessary for relating actions with each other.)
```
I'm not really understanding this one @mckrava, can you please try to rephrase it and then ping me?

Reply:
`op_id` - A unique operation identifier. It should have the same value for all actions of one operation/action (the same `op_id` value for  one chain of Domain Registration actions `DMN_REG` -> `DMN_REG_OK` || `DMN_REG_REFUND`  It's necessary to build relationship between actions of a single purchase order.)

@Yung-Beef

Reply:
```suggestion
`op_id` - A unique operation identifier. It should have the same value for all instances of a particular type of operation/action (the same `op_id` value for one chain of Domain Registration actions `DMN_REG` -> `DMN_REG_OK` || `DMN_REG_REFUND` It's necessary to build relationship between actions of a single purchase order.)
```
The first part is good now, but the stuff in the parentheses is confusing me... the first part says to use the same value for all instances of a particular type of operation, but then there are 3 operations mentioned in the parentheses?

Reply:
Ah, ok. Let me explain an idea - we have such action like `Domain Generation Request` (`DMN_REG`). UI app (Polkaverse) creates registration request (system remark) and UI app adds unique ID (`op_id === 0x123`) for this particular request.  Then our seller-squid catches this request and creates `DomainRegistrationOrder` with ID === `op_id` === 0x123. Then, if registration is successfully done by seller-squid, it sends to Polkadot chain another remark as confirmation of successful registration (`DMN_REG_OK`) or refund (`DMN_REG_REFUND`) request. Both of these requests/remarks must have op_id === 0x123  to understand that this particular confirmation or refund request belongs to specific DomainRegistrationOrder with ID === 0x123. 
I hope it should help to compile human-readable explanation )

@Yung-Beef

Reply:
```suggestion
`op_id` - A unique operation identifier that is used to build a relationship between Domain Registration actions, in order to associate a particular `DMN_REG_OK` or `DMN_REG_REFUND` to the corresponding `DMN_REG`.
```

How does that look? 

If the || means "or", you can switch it back to || if you think it would be easier for devs to read.